resources/sql/application-queries.sql

```diff
@@ -613,46 +620,87 @@ JOIN virkailija_credentials AS vc ON vc.application_key = a.key
 WHERE vc.secret = :virkailija_secret;
 
 -- name: yesql-get-latest-version-by-secret-lock-for-update
-WITH latest_version AS (
-    SELECT max(created_time) AS latest_time
-    FROM applications a
-    WHERE a.secret = :secret
-)
-SELECT
-  id,
-  key,
-  lang,
-  form_id AS form,
-  created_time,
-  content,
-  haku,
-  hakukohde,
-  person_oid
+WITH application_secret AS (SELECT
+                              application_key,
+                              id
+                            FROM application_secrets
+                            WHERE secret = :secret
+                            ORDER BY id DESC
+                            LIMIT 1),
+    latest_secret AS (SELECT
+                        application_key,
+                        id
+                      FROM application_secrets
+                      WHERE application_key = (SELECT application_key
+                                               FROM application_secret)
+                      ORDER BY id DESC
+                      LIMIT 1)
+SELECT *
 FROM applications a
-  JOIN latest_version lv ON a.created_time = lv.latest_time
+  JOIN application_secret ON application_secret.application_key = a.key
+  JOIN latest_secret ON latest_secret.application_key = a.key
+WHERE application_secret.id = latest_secret.id
+ORDER BY a.id DESC
+LIMIT 1
 FOR UPDATE;
 
+-- name: yesql-get-application-count-for-secret
+SELECT count(id) AS count
+FROM application_secrets
+WHERE secret = :secret;
+
+-- name: yesql-get-application-key-for-any-version-of-secret
+SELECT application_key
+FROM application_secrets
+WHERE secret = :secret
+ORDER BY id DESC
+LIMIT 1;
+
+-- name: yesql-get-latest-application-language-by-any-version-of-secret
+WITH application_secret AS (SELECT DISTINCT ON (application_key) *
+                            FROM application_secrets
+                            WHERE secret = :secret
+                            ORDER BY application_key, id DESC)
+SELECT lang
+FROM latest_applications
+WHERE key = (SELECT application_key
```

Comment:
Tässä riittäis kanssa `WHERE key = (SELECT application_key FROM application_secret WHERE secret = :secret)`